Engine/Helper.cs

```diff
@@ -1602,6 +1602,135 @@ public bool GetNamedArgumentAttributeValue(NamedAttributeArgumentAst namedAttrib
+                    "PrivateData",
+                    "HelpInfoURI",
+                    "DefaultCommandPrefix"};
+            keys.AddRange(keysCommon);
+            if (powershellVersion.Equals(new Version("5.0")))
+            {
+                keys.Add("DscResourcesToExport");
+            }
+            else if (!powershellVersion.Equals(new Version("4.0"))
+                && !powershellVersion.Equals(new Version("3.0")))
+            {
+                throw new ArgumentException("Invalid PowerShell version. Choose from 3.0, 4.0 or 5.0");
+            }
+            return keys;
+        }
+
+        /// <summary>
+        /// Gets deprecated keys of PowerShell module manifest
+        /// </summary>
+        /// <returns>Returns an enumerator over deprecated keys</returns>
+        public static IEnumerable<string> GetDeprecatedModuleManifestKeys()
+        {
+            return new List<string> { "ModuleToProcess" };
+        }
+
+        /// <summary>
+        /// Checks if a given file is a valid PowerShell module manifest
+        /// </summary>
+        /// <param name="filepath">Path to module manifest</param>
+        /// <returns>true if given filepath points to a module manifest, otherwise false</returns>
+        public static bool IsModuleManifest(string filepath)
+        {
+            // 4.0 and 3.0 contain the same keys. Hence, compare only with 4.0.
+            return IsModuleManifest(filepath, new Version("5.0"))
+                || IsModuleManifest(filepath, new Version("4.0"));
+        }
+
+        /// <summary>
+        /// Checks if a given file is a valid PowerShell module manifest
+        /// </summary>
+        /// <param name="filepath">Path to module manifest</param>
+        /// <param name="powershellVersion">Version parameter with valid values: 5.0, 4.0 and 3.0</param>
+        /// <returns>true if given filepath points to a module manifest, otherwise false</returns>
+        public static bool IsModuleManifest(string filepath, Version powershellVersion)
+        {
+            Token[] tokens;
+            ParseError[] errors;
+            if (filepath == null)
+            {
+                throw new ArgumentNullException("filepath");
+            }
+            if (powershellVersion == null)
+            {
+                throw new ArgumentNullException("powershellVersion");
+            }
+            if (!Path.GetExtension(filepath).Equals(".psd1", StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+
+            //using parsefile causes the parser to crash!
+            string fileContent = File.ReadAllText(filepath);
+            var ast = Parser.ParseInput(fileContent, out tokens, out errors);
+            var hast = ast.Find(x => x is HashtableAst, false) as HashtableAst;
+            if (hast == null)
+            {
+                return false;
+            }
+            var validKeys = GetModuleManifestKeys(powershellVersion);
```

Comment:
>powershellVersion [](start = 50, length = 17)

I think, we should read the PowerShellVersion from PSD1 and validate validate the manifest keys.
PowerShellVersion is minimum required PowerShell version for importing the module.